File: src/ClickThroughPlayerView.m
#import "ClickThroughPlayerView.h"

@implementation ClickThroughPlayerView

- (instancetype)initWithFrame:(NSRect)frameRect {
    self = [super initWithFrame:frameRect];
    if (self) {
        self.controlsStyle = AVPlayerViewControlsStyleNone;
        self.videoGravity = AVLayerVideoGravityResizeAspectFill;
        self.updatesNowPlayingInfoCenter = NO;
    }
    return self;
}

// Pass all mouse events directly through to Finder / desktop icons
- (NSView *)hitTest:(NSPoint)point {
    return nil;
}

@end

File: swift_src/ClickThroughPlayerView.swift
import Cocoa
import AVKit

/// Custom AVPlayerView that passes all mouse clicks directly through to Finder desktop icons
final class ClickThroughPlayerView: AVPlayerView {
    override init(frame frameRect: NSRect) {
        super.init(frame: frameRect)
        self.controlsStyle = .none
        self.videoGravity = .resizeAspectFill
        self.updatesNowPlayingInfoCenter = false // Prevent hijacking media keys / Control Center
    }

    required init?(coder: NSCoder) {
        super.init(coder: coder)
        self.controlsStyle = .none
        self.videoGravity = .resizeAspectFill
        self.updatesNowPlayingInfoCenter = false
    }

    /// Returning nil ensures mouse events bypass this view and reach desktop icons
    override func hitTest(_ point: NSPoint) -> NSView? {
        return nil
    }
}
